File: src/main/java/com/jacuum/web/AlgosEndpoint.java
package com.jacuum.web;

import com.jacuum.algo.Algorithms;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import java.util.List;

@RestController
@RequestMapping("/api")
public final class AlgosEndpoint {

    private final Algorithms algorithms;

    public AlgosEndpoint(final Algorithms algorithms) {
        this.algorithms = algorithms;
    }

    @GetMapping("/algos")
    public List<String> algos() {
        return this.algorithms.names();
    }

    @GetMapping("/avatars")
    public List<String> avatars() {
        return List.of("🤖", "🦾", "👾", "🚀", "🛸", "🦄", "🐢", "🦊", "🐱", "🐸");
    }
}

File: src/main/kotlin/com/jacuum/web/AlgosEndpoint.kt
package com.jacuum.web

import com.jacuum.algo.Algorithms
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api")
class AlgosEndpoint(private val algorithms: Algorithms) : AlgosApi {
    @GetMapping("/algos")
    override fun algos(): List<String> = algorithms.names()

    @GetMapping("/avatars")
    override fun avatars(): List<String> =
        listOf("\uD83E\uDD16", "\uD83E\uDDBE", "\uD83D\uDC7E", "\uD83D\uDE80",
               "\uD83D\uDEF8", "\uD83E\uDD84", "\uD83D\uDC22", "\uD83E\uDD8A",
               "\uD83D\uDC31", "\uD83D\uDC38")
}
